README.md

```diff
@@ -4,6 +4,64 @@ This is a REST-style API that uses JSON for serialization and [API token](#authe
 
 ## Authentication
 
+### OAuth 2 Authentication
+
+This is our preferred way of allowing users to grant access to their accounts from third party applications. This way our users have complete control over access to their data without giving out their password.
+
+1. [Grab an OAuth 2 library](http://oauth.net/code/)
+2. Contact support to register your application and provide us with a `redirect_uri`. You will then be assigned a `client_id` and `client_secret`. (The `redirect_uri` is where we will send the verification code.)
+3. Configure your OAuth 2 library with:
+  * `client_id`, `client_secret`, and `redirect_uri`
+  * `https://app.outstand.com/oauth/authorize` to request authorization
+  * `https://app.outstand.com/oauth/token` to get access tokens
+4. Try making an authorized request to `https://app.outstand.com/oauth/token/info`
+
+If you are **not** using a library, you will need to request access:
+
+```
+https://app.outstand.com/oauth/authorize?client_id=[client_id]&redirect_uri=[redirect_uri]&response_type=code
+```
+
+We will then authenticate the user, ask them to authorize your application, and redirect the user back to your application with an authorization code. (This authorization code is good for 5 minutes.)
+
+Your application must then make a request to trade your authorization code for an access token:
+
+```
+POST https://app.outstand.com/oauth/token
+```
+
+with a body of:
+```json
+{
+  "client_id": [client_id],
+  "client_secret": [client_secret],
+  "code": [authorization code],
+  "grant_type": "authorization_code",
+  "redirect_uri": "https://example.com/auth/outstand/callback"
+}
+```
+
+You'll receive a `200 OK` with the following response body:
+
+```json
+{
+  "access_token": "...",
+  "token_type": "bearer",
+  "expires_in": 7200,
+  "refresh_token": "...",
+  "created_at": 1447820578
+}
+```
+
+Your application will then use the `access_token` with your API requests by setting the Authorization header:
+
+```
+Authorization: Bearer ACCESS_TOKEN
```

Comment:
We need to include information about our use of refresh tokens and what the access token timeout is (currently 2 hours).